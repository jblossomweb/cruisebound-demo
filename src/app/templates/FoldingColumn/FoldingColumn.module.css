.page {
  @apply w-full h-screen flex flex-col md:flex-row;
}

.left {
  @apply w-screen md:w-16
    h-16 md:h-screen
    bg-sky-950 text-gray-200;
  transition: width 0.5s, height 0.5s;
}

.left .main {
  @apply opacity-0;
  transition: opacity 0.25s;
}

.left.expanded {
  @apply w-screen h-96
  md:w-80 md:h-screen;
}

.left .toggle button img,
.left .toggle button svg {
  @apply origin-center
  rotate-90
  md:rotate-0;
}

.left.expanded .toggle button img,
.left.expanded .toggle button svg {
  @apply origin-center
  -rotate-90
  md:-rotate-180;
}

.left.expanded .main {
  @apply opacity-100;
}

.right {
  @apply h-screen w-full overflow-y-auto bg-white text-black;
}

.content {
  @apply p-4 w-full h-full;
}

.stack {
  @apply w-full h-full flex flex-col justify-between;
}

.top {
  @apply w-full flex flex-col justify-start;
}

.toggle {
  @apply w-full text-right pb-4;
}

.toggle button {
  @apply p-2 leading-3;
}

.toggle button img,
.toggle button svg {
  transition: transform 0.5s;
}

.footer {
  @apply w-full text-center;
}
