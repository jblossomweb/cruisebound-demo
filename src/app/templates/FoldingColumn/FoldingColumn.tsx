import React, { useState } from 'react'
import classNames from 'classnames'

import Button from 'app/components/atoms/Button'
import Image from 'app/components/atoms/Image'
import Arrow from 'app/components/atoms/Arrow'
import logo from 'svg/logo.svg'

import styles from './FoldingColumn.module.css'

export interface FoldingColumnProps {
  leftContent: React.ReactNode
  rightContent: React.ReactNode
}

const FoldingColumn: React.FC<FoldingColumnProps> = ({
  leftContent,
  rightContent,
}) => {
  const [expanded, setExpanded] = useState(true)

  return (
    <div className={styles.page}>
      <div
        data-testid="left-col"
        className={classNames(styles.left, { [styles.expanded]: expanded })}
      >
        <div className={styles.content}>
          <div className={styles.stack}>
            <div className={styles.top}>
              <div className={styles.toggle}>
                <Button
                  text={<Arrow color="#fff" size={12} />}
                  color="slate"
                  onClick={() => {
                    setExpanded(!expanded)
                  }}
                />
              </div>
              <div className={styles.main}>{leftContent}</div>
            </div>
            <div className={styles.footer}>
              <Image src={logo} width={176} height={86} alt="Cruisebound" />
            </div>
          </div>
        </div>
      </div>
      <div className={styles.right}>
        <div className={styles.content}>{rightContent}</div>
      </div>
    </div>
  )
}

export default FoldingColumn
