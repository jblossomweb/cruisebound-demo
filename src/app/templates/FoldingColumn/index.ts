export type { ButtonProps } from './FoldingColumn'
export { default } from './FoldingColumn'
