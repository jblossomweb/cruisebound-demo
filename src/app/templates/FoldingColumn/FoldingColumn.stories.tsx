import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import FoldingColumn, { type FoldingColumnProps } from './FoldingColumn'

export default {
  title: 'templates/FoldingColumn',
  component: FoldingColumn,
  argTypes: {
    text: { control: 'text' },
    color: { control: 'select', options: ['blue', 'red', 'green'] },
    size: { control: 'select', options: ['small', 'medium', 'large'] },
  },
} as Meta<typeof FoldingColumn>

const Template: StoryFn<typeof FoldingColumn> = (args: FoldingColumnProps) => (
  <FoldingColumn {...args} />
)

export const Sample = Template.bind({})
Sample.args = {
  leftContent: 'left side content',
  rightContent: 'right side content',
}
