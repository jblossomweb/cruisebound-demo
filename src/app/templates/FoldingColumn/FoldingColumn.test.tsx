/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import FoldingColumn, { type FoldingColumnProps } from './FoldingColumn'
import styles from './FoldingColumn.module.css'
import * as stories from './FoldingColumn.stories'

describe('app/templates/FoldingColumn', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('interaction', () => {
    afterEach(cleanup)

    it('should start out as expanded', () => {
      const { asFragment } = render(
        <FoldingColumn {...(stories.Sample.args as FoldingColumnProps)} />
      )
      expect(asFragment()).toMatchSnapshot()
      const leftColBefore = screen.getByTestId('left-col')
      expect(leftColBefore.className).toContain(styles.expanded)
    })

    it('should collapse left column when arrow is clicked', () => {
      const { asFragment } = render(
        <FoldingColumn {...(stories.Sample.args as FoldingColumnProps)} />
      )
      const button = screen.getByRole('button')
      expect(asFragment()).toMatchSnapshot()
      const leftColBefore = screen.getByTestId('left-col')
      expect(leftColBefore.className).toContain(styles.expanded)
      fireEvent.click(button)
      expect(asFragment()).toMatchSnapshot()
      const leftColAfter = screen.getByTestId('left-col')
      expect(leftColAfter.className).not.toContain(styles.expanded)
    })
  })
})
