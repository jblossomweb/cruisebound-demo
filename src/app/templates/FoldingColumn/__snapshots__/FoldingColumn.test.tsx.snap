// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/templates/FoldingColumn interaction should collapse left column when arrow is clicked 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="left expanded"
      data-testid="left-col"
    >
      <div
        class="content"
      >
        <div
          class="stack"
        >
          <div
            class="top"
          >
            <div
              class="toggle"
            >
              <button
                class="btn btn-slate btn-medium"
                type="button"
              >
                <svg
                  fill="none"
                  height="12"
                  viewBox="0 0 9 9"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 4.5L8.5 4.5"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M4.75 0.750001L8.5 4.5L4.75 8.25"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="main"
            >
              left side content
            </div>
          </div>
          <div
            class="footer"
          >
            <figure
              class=""
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2786%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Cruisebound"
                  aria-label="Cruisebound"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  title="Cruisebound"
                />
                <noscript />
              </span>
            </figure>
          </div>
        </div>
      </div>
    </div>
    <div
      class="right"
    >
      <div
        class="content"
      >
        right side content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/templates/FoldingColumn interaction should collapse left column when arrow is clicked 2`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="left"
      data-testid="left-col"
    >
      <div
        class="content"
      >
        <div
          class="stack"
        >
          <div
            class="top"
          >
            <div
              class="toggle"
            >
              <button
                class="btn btn-slate btn-medium"
                type="button"
              >
                <svg
                  fill="none"
                  height="12"
                  viewBox="0 0 9 9"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 4.5L8.5 4.5"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M4.75 0.750001L8.5 4.5L4.75 8.25"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="main"
            >
              left side content
            </div>
          </div>
          <div
            class="footer"
          >
            <figure
              class=""
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2786%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Cruisebound"
                  aria-label="Cruisebound"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  title="Cruisebound"
                />
                <noscript />
              </span>
            </figure>
          </div>
        </div>
      </div>
    </div>
    <div
      class="right"
    >
      <div
        class="content"
      >
        right side content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/templates/FoldingColumn interaction should start out as expanded 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="left expanded"
      data-testid="left-col"
    >
      <div
        class="content"
      >
        <div
          class="stack"
        >
          <div
            class="top"
          >
            <div
              class="toggle"
            >
              <button
                class="btn btn-slate btn-medium"
                type="button"
              >
                <svg
                  fill="none"
                  height="12"
                  viewBox="0 0 9 9"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 4.5L8.5 4.5"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M4.75 0.750001L8.5 4.5L4.75 8.25"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="main"
            >
              left side content
            </div>
          </div>
          <div
            class="footer"
          >
            <figure
              class=""
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2786%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Cruisebound"
                  aria-label="Cruisebound"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  title="Cruisebound"
                />
                <noscript />
              </span>
            </figure>
          </div>
        </div>
      </div>
    </div>
    <div
      class="right"
    >
      <div
        class="content"
      >
        right side content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/templates/FoldingColumn stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="left expanded"
      data-testid="left-col"
    >
      <div
        class="content"
      >
        <div
          class="stack"
        >
          <div
            class="top"
          >
            <div
              class="toggle"
            >
              <button
                class="btn btn-slate btn-medium"
                type="button"
              >
                <svg
                  fill="none"
                  height="12"
                  viewBox="0 0 9 9"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 4.5L8.5 4.5"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M4.75 0.750001L8.5 4.5L4.75 8.25"
                    stroke="#fff"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
            <div
              class="main"
            >
              left side content
            </div>
          </div>
          <div
            class="footer"
          >
            <figure
              class=""
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2786%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Cruisebound"
                  aria-label="Cruisebound"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  title="Cruisebound"
                />
                <noscript />
              </span>
            </figure>
          </div>
        </div>
      </div>
    </div>
    <div
      class="right"
    >
      <div
        class="content"
      >
        right side content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
